### PR TITLE
Convert assessment group deletion sproc to library function

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Also fixes an error in the instructor assessment instance page that caused it to crash if the group was deleted (https://github.com/PrairieLearn/PrairieLearn/issues/6541#issuecomment-1313680436).

### CODE REVIEW

File: apps/prairielearn/src/lib/groups.ts [hunk 1]
```diff
@@ -600,6 +600,17 @@ export async function updateGroupRoles(
...
 }
 
+export async function deleteGroup(assessment_id: string, group_id: string, authn_user_id: string) {
+  const deleted_group_id = await sqldb.queryOptionalRow(
+    sql.delete_group,
+    { assessment_id, group_id, authn_user_id },
+    IdSchema,
+  );
+  if (deleted_group_id == null) {
+    throw new Error('Group does not exist.');
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/groups.ts:610]: 403/404 instead?

File: apps/prairielearn/src/lib/groups.sql [hunk 2]
```diff
@@ -342,6 +342,72 @@ WHERE
...
+      group_logs (authn_user_id, group_id, action)
+    SELECT
+      $authn_user_id,
+      id,
+      'delete'
+    FROM
+      deleted_group
+  )
+SELECT
+  *
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/groups.sql:407]: Nit, I'd rather be explicit:

```suggestion
  id
```

**Review by nwalters512 — APPROVED** [review]
Looks reasonable 👍

File: apps/prairielearn/src/lib/groups.ts (hunk 1 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/groups.ts]: Done in 74af8894a6b0571eac1c04f2d95ef0e31fe4437b.

**Pull request merged**